### DISCUSSION TITLE
Having error with my endpoints base on 'HTTP'

### DISCUSSION THREAD

**Discussion by connelevalsam** [vercel/community#683] [Help] [answered]
Hello good day all,

I have deployed my NextJS app on vercel, but when I try to visit the page making use of the endpoint I get the: 
```
football-80a32d4130451e3b.js:1 Mixed Content: The page at 'https://abatino.vercel.app/football' was loaded over HTTPS, but 

requested an insecure resource 'http://38.242.211.41:5010/WeatherForecast/GetTodaysGames'. This request has been blocked; the 

content must be served over HTTPS.
```

Then I added in my `<HEAD>` this:
```

<meta


          http-equiv="Content-Security-Policy"
          content="upgrade-insecure-requests"
        

></meta>
```

I rebuilt it and now I am having this:
`
net::ERR_SSL_PROTOCOL_ERROR
`

I don't know what next to do

**Comment by swarnava** [comment, marked as answer]
As indicated in the error, mixed content occurs when you're requesting content over HTTP on a page that's already been loaded over HTTPS. You need to make sure that the resources are requested over HTTPS instead of HTTP by changing the URL's to reflect this.

You can find more information on this here: https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content/How_to_fix_website_with_mixed_content

**Reply by connelevalsam** [reply]
> As indicated in the error, mixed content occurs when you're requesting content over HTTP on a page that's already been loaded over HTTPS. You need to make sure that the resources are requested over HTTPS instead of HTTP by changing the URL's to reflect this.
> 
> You can find more information on this here: https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content/How_to_fix_website_with_mixed_content

Thanks for your answer. What if I can't control where the API reads from?
Is there no way I'd be able to make it work?

**Reply by swarnava** [reply]
Unfortunately not. Most of the web browser expect HTTPS protocol by  default in most cases. There might be some hacky approach we won't recommend. Thanks!

**Reply by connelevalsam** [reply]
Alright thanks for your time
